src/Input/NumberInput/NumberInput.test.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+
+import * as renderer from 'react-test-renderer';
+import '@testing-library/jest-dom/extend-expect';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import NumberInput from './NumberInput';
+
+const props = {
+  label: 'Number',
+  onChange: jest.fn().mockImplementation(event => event.target.value),
+};
+
+const inputValue = '12';
+
+function setupNumberInput(otherProps?: any) {
+  const { getByLabelText, ...utils } = render(
+    <NumberInput
+      label={props.label}
+      onChange={props.onChange}
+      {...otherProps}
+    />
+  );
+  const numberInput = getByLabelText(props.label) as HTMLInputElement;
+  return { numberInput, utils };
+}
+
+function setupNumberInputWithChangeEvent(otherProps?: any) {
+  const { numberInput } = setupNumberInput(otherProps);
+  userEvent.type(numberInput, inputValue);
+  return { numberInput };
+}
+
+it('<NumberInput> should render with a label of Username', () => {
+  const TextFieldSnapshot = renderer
+    .create(<NumberInput label={props.label} />)
+    .toJSON();
+  expect(TextFieldSnapshot).toMatchSnapshot();
+});
+
+describe('when it is rendered', () => {
+  it('should have an empty string as its value', () => {
```

Comment:
I guess we can add more context to the test case like `describe('when it is rendered without value passed to it' `